setup.sh

```diff
@@ -0,0 +1,282 @@
+  b | branch) BRANCH="$OPTARG" ;;
+  d | dir) DIR="$OPTARG" ;;
+  g | git-repo) GIT_REPO="$OPTARG" ;;
+  r | runpod) RUNPOD=true ;;
+  h) display_help && exit 0 ;;
+  *) display_help && exit 0 ;;
+  esac
+done
+shift $((OPTIND - 1))
+
+# This must be set after the getopts loop to account for $DIR changes.
+PARENT_DIR="$(dirname "${DIR}")"
+VENV_DIR="$DIR/venv"
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  # Check if root or sudo
+  root=true
+  if [ "$EUID" -ne 0 ]; then
+    root=false
+  fi
+
+  if env_var_exists RUNPOD_POD_ID || env_var_exists RUNPOD_API_KEY; then
+    RUNPOD=true
+  fi
+
+  env_var_exists() {
+    local env_var=
+    env_var=$(declare -p "$1")
+    if ! [[ -v $1 && $env_var =~ ^declare\ -x ]]; then
+      return 1
+    fi
+  }
+
+  get_distro_name() {
+    local line
+    if [ -f /etc/os-release ]; then
+      # We search for the line starting with ID=
+      # Then we remove the ID= prefix to get the name itself
+      line="$(grep -Ei '^ID=' /etc/os-release)"
+      line=${line##*=}
+      echo "$line"
+      return 0
+    elif command -v python >/dev/null; then
+      line="$(python -mplatform)"
+      echo "$line"
+      return 0
+    elif command -v python3 >/dev/null; then
+      line="$(python3 -mplatform)"
+      echo "$line"
+      return 0
+    else
+      line="None"
+      echo "$line"
+      return 1
+    fi
+  }
+
+  get_distro_family() {
+    local line
+    if [ -f /etc/os-release ]; then
+      # We search for the line starting with ID_LIKE=
+      # Then we remove the ID_LIKE= prefix to get the name itself
+      # This is the "type" of distro. For example, Ubuntu returns "debian".
+      if grep -Eiq '^ID_LIKE=' /etc/os-release >/dev/null; then
+        line="$(grep -Ei '^ID_LIKE=' /etc/os-release)"
+        line=${line##*=}
+        echo "$line"
+        return 0
+      else
+        line="None"
+        echo "$line"
+        return 1
+      fi
+    else
+      line="None"
+      echo "$line"
+      return 1
+    fi
+  }
+
+  # This is the pre-install work for a kohya installation on a runpod
+  if [ "$RUNPOD" = true ]; then
+    if [ -d "$VENV_DIR" ]; then
+      echo "Pre-existing installation on a runpod detected."
+      export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:"$VENV_DIR"/lib/python3.10/site-packages/tensorrt/
+      export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:"$VENV_DIR"/lib/python3.10/site-packages/nvidia/cuda_runtime/lib/
+      cd "$DIR" || exit 1
+      sed -i "s/interface.launch(\*\*launch_kwargs)/interface.launch(\*\*launch_kwargs,share=True)/g" ./kohya_gui.py
+    else
+      echo "Clean installation on a runpod detected."
+      cd "$PARENT_DIR" || exit 1
+      if [ ! -d "$DIR/.git" ]; then
+        echo "Cloning $GIT_REPO."
+        git clone "$GIT_REPO"
+        cd "$DIR" || exit 1
+        git checkout "$BRANCH"
+      else
+        cd "$DIR" || exit 1
+        echo "git repo detected. Attempting tp update repo instead."
+        echo "Updating: $GIT_REPO"
+        git pull "$GIT_REPO"
```

Comment:
I'd be hesitant to update repo every time, since we don't update dependencies etc. We also have `upgrade.sh` script.

Reply:
So, I did that only if we detect a git repo in the configured installation folder, but we do not detect a venv folder. I'm assuming a busted or new environment at that point which is why I did it this way. The installation script down below should be installing the pip requirements.

Reply:
My bad, I missed the check if venv doesn't exist.